server/src/services/folders.service.ts

```diff
@@ -31,8 +31,8 @@ export class FoldersService {
     );
   }
 
-  async all(user: User): Promise<Folder[]> {
-    if (await this.usersService.hasGroup(user, 'admin')) {
+  async all(user: User, searchKey: string): Promise<Folder[]> {
```

Comment:
Can we extract out logic for getting folders when search is involved onto a separate function? This seems complex as of now.